backend/src/main/java/com/votogether/domain/member/entity/Member.java

```diff
@@ -77,10 +80,21 @@ public static Member from(final KakaoMemberResponse response) {
                 .build();
     }
 
-    public void changeNickname(final String nickname) {
+    public void changeNicknameByCycle(final String nickname, final Long days) {
+        if (nickname.startsWith(INITIAL_NICKNAME_PREFIX)) {
+            throw new BadRequestException(MemberExceptionType.INVALID_NICKNAME_LETTER);
+        }
+        if (isNotPassedChangingCycle(days)) {
+            throw new BadRequestException(MemberExceptionType.NOT_PASSED_NICKNAME_CHANGING_CYCLE);
+        }
         this.nickname = new Nickname(nickname);
     }
 
+    private boolean isNotPassedChangingCycle(final Long days) {
+        return (this.nickname.nonStartsWith(INITIAL_NICKNAME_PREFIX)) &&
+                (this.getUpdatedAt().isAfter(LocalDateTime.now().minusDays(days)));
+    }
+
```

Comment:
`P2`
`isNotPassedChangingCycle` 메서드에 프리픽스로 시작하는 닉네임인지를 검증하는게 어색하다고 생각되는데
&& 로 연결 되어있는 두 boolen 식을 각각 메서드로 추출하고
추출한 메서드를 `changeNicknameByCycle`에서 &&로 연결한는건 어떤가요??


```suggestion
    public void changeNicknameByCycle(final String nickname, final Long days) {
        if (nickname.startsWith(INITIAL_NICKNAME_PREFIX)) {
            throw new BadRequestException(MemberExceptionType.INVALID_NICKNAME_LETTER);
        }
        if (isNotPassedChangingCycle(days) && ! isInitalNickname()) {
            throw new BadRequestException(MemberExceptionType.NOT_PASSED_NICKNAME_CHANGING_CYCLE);
        }
        this.nickname = new Nickname(nickname);
    }

    private boolean isNotPassedChangingCycle(final Long days) {
        return this.getUpdatedAt().isAfter(LocalDateTime.now().minusDays(days));
    }
    
    private boolean isInitalNickname() {
       return this.nickname.startsWith(INITIAL_NICKNAME_PREFIX) 
    }

```

Reply:
좋은 제안인 것 같아요 ㅎㅎ
안그래도 함수의 역할이 명확하지 않다는 생각만하고 그냥 넘어갔던 것 같은데 짚어주셔서 감사합니다!
말씀하신 방향대로 수정해볼게요 :)